src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -1,38 +1,114 @@
+}
+
+export interface AlertInternal extends Alert {
+    closeFunction?: (callback: () => void) => void;
+    isOpen: boolean;
 }
 
 @Injectable({
     providedIn: 'root',
 })
 export class AlertService {
     timeout = 8000;
-    toast = false;
-    position = 'top right';
+    dismissible = true;
 
     // unique id for each alert. Starts from 0.
     private alertId = 0;
-    private alerts: Alert[] = [];
+    private alerts: AlertInternal[] = [];
 
-    constructor(private sanitizer: DomSanitizer, private ngZone: NgZone, private translateService: TranslateService) {}
+    errorListener: Subscription;
+    httpErrorListener: Subscription;
+
+    constructor(private sanitizer: DomSanitizer, private ngZone: NgZone, private translateService: TranslateService, private eventManager: EventManager) {
+        this.errorListener = eventManager.subscribe('artemisApp.error', (response: EventWithContent<unknown> | string) => {
+            const errorResponse = (response as EventWithContent<AlertError>).content;
+            this.addErrorAlert(errorResponse.message, errorResponse.translationKey, errorResponse.translationParams);
+        });
+
+        this.httpErrorListener = eventManager.subscribe('artemisApp.httpError', (response: any) => {
+            const httpErrorResponse = response.content;
+            switch (httpErrorResponse.status) {
+                // connection refused, server not reachable
+                case 0:
+                    this.addErrorAlert('Server not reachable', 'error.server.not.reachable');
+                    break;
+
+                case 400:
+                    if (httpErrorResponse.error !== null && httpErrorResponse.error.skipAlert) {
+                        break;
+                    }
+                    const arr = httpErrorResponse.headers.keys();
+                    let errorHeader = null;
+                    let entityKey = null;
+                    arr.forEach((entry: string) => {
+                        if (entry.toLowerCase().endsWith('app-error')) {
+                            errorHeader = httpErrorResponse.headers.get(entry);
+                        } else if (entry.toLowerCase().endsWith('app-params')) {
+                            entityKey = httpErrorResponse.headers.get(entry);
+                        }
+                    });
+                    if (errorHeader) {
+                        const entityName = translateService.instant('global.menu.entities.' + entityKey);
+                        this.addErrorAlert(errorHeader, errorHeader, { entityName });
+                    } else if (httpErrorResponse.error && httpErrorResponse.error.fieldErrors) {
+                        const fieldErrors = httpErrorResponse.error.fieldErrors;
+                        for (const fieldError of fieldErrors) {
+                            if (['Min', 'Max', 'DecimalMin', 'DecimalMax'].includes(fieldError.message)) {
+                                fieldError.message = 'Size';
```

Comment:
To be honest: I have no idea what this is doing. I've copied this code from the old error component implementation and just adjusted the calls to the alert service. I assume that this is somehow related to server side validation of min and max boundaries of whatever numeric parameters (as comes up if you quickly google 'DecimalMin java', for example).

Reply:
True, I suppose it gentrifies the error message => if you have a field with min and max, it only tells you that something with the size is wrong. Probably because we don't know which side has the issue.

Would be great if you could add one sentence regarding that.

Reply:
I've added a small comment for it!